api/config/config.go

```diff
@@ -403,35 +420,209 @@ func (u *FeastServingURLs) URLs() []string {
 }
 
 type JaegerConfig struct {
-	AgentHost    string `envconfig:"JAEGER_AGENT_HOST"`
-	AgentPort    string `envconfig:"JAEGER_AGENT_PORT"`
-	SamplerType  string `envconfig:"JAEGER_SAMPLER_TYPE" default:"probabilistic"`
-	SamplerParam string `envconfig:"JAEGER_SAMPLER_PARAM" default:"0.01"`
-	Disabled     string `envconfig:"JAEGER_DISABLED" default:"true"`
+	AgentHost    string
+	AgentPort    string
+	SamplerType  string `validate:"required" default:"probabilistic"`
+	SamplerParam string `validate:"required" default:"0.01"`
+	Disabled     string `validate:"required" default:"true"`
 }
 
 type MlflowConfig struct {
-	TrackingURL string `envconfig:"MLFLOW_TRACKING_URL" required:"true"`
+	TrackingURL string `validate:"required"`
 }
 
-func InitConfigEnv() (*Config, error) {
-	var cfg Config
-	if err := envconfig.Process("", &cfg); err != nil {
-		return nil, err
+func (cfg *Config) Validate() error {
+	v := validator.New()
+	err := v.Struct(cfg)
+	if err != nil {
+		return err
 	}
-	cfg.EnvironmentConfigs = initEnvironmentConfigs(cfg.EnvironmentConfigPath)
-	if err := cfg.validate(); err != nil {
-		return nil, err
+	// Validate pyfunc server keep alive config, it must be string in json format
+	var pyfuncGRPCOpts json.RawMessage
+	if err := json.Unmarshal([]byte(cfg.PyfuncGRPCOptions), &pyfuncGRPCOpts); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Load creates a Config object from default config values, config files and environment variables.
+// Load accepts config files as the argument. JSON and YAML format are both supported.
+//
+// If multiple config files are provided, the subsequent config files will override the config
+// values from the config files loaded earlier.
+//
+// These config files will override the default config values (refer to setDefaultValues function)
+// and can be overridden by the values from environment variables. Nested keys in the config
+// can be set from environment variable name separated by "_". For instance the config value for
+// "DbConfig.Port" can be overridden by environment variable name "DBCONFIG_PORT". Note that
+// all environment variable names must be upper case.
+//
+// If no config file is provided, only the default config values and config values from environment
+// variables will be loaded.
+//
+// Refer to example.yaml for an example of config file.
```

Comment:
Thanks for detailed comment!

Reply:
Ahaha I just copied that entire chunk of code from Turing. In the long run we'll remove this entire function into a separate helper function in the MLP package so that Merlin, Turing and XP can all use it.